plugins/vlc-video/vlc-video-source.c

```diff
@@ -394,12 +394,97 @@ static int vlcs_audio_setup(void **p_data, char *format, unsigned *rate,
 	return 0;
 }
 
+static inline void vlc_split_media_path(const struct dstr *new_path,
+	struct dstr *path_only, struct dstr *options)
+{
+	int i;
+	const char* path = new_path->array;
+	bool quoting = false;
+	int options_at = 0;
+	int path_len;
+
+	for (i = 0; path[i]; ++i) {
+		if (path[i] == '"') {
+			quoting = !quoting;
+		}
+		if (!quoting && path[i+1]==' ' && path[i+2]==':') {
+			// ++i to be consistent with finished loop case
+			options_at = ++i;
+			break;
+		}
+	}
+	if (options_at) { // grab options
+		dstr_copy(options, new_path->array+options_at);
+	}
+	for(--i; path[i]==' '; --i); // remove trailing spaces
+	path_len = i+1;
+	dstr_ncopy_dstr(path_only, new_path, path_len);
+}
+
+static inline void vlc_apply_media_options(libvlc_media_t *media,
+	const char *options, const bool is_url)
+{
+	int i, j;
+	bool quoting = false;
+	int option_at = 0;
+	int option_len;
+	char* option_str = 0;
```

Comment:
Style: The asterisk should be next to the variable name when declaring a pointer. Use "char *option_str".